src/Tribe/Repositories/Attendee/RSVP.php

```diff
@@ -77,4 +112,109 @@ public function checked_in_keys() {
 		return $this->limit_list( $this->key_name, parent::checked_in_keys() );
 	}
 
+	/**
+	 * Filter the arguments to set for the attendee for this provider.
+	 *
+	 * @since TBD
+	 *
+	 * @param array                         $args          List of arguments to set for the attendee.
+	 * @param array                         $attendee_data List of additional attendee data.
+	 * @param Tribe__Tickets__Ticket_Object $ticket        The ticket object or null if not relying on it.
+	 *
+	 * @return array List of arguments to set for the attendee.
+	 */
+	public function filter_set_attendee_args( $args, $attendee_data, $ticket = null ) {
+		// Set default order ID.
+		if ( empty( $args['order_id'] ) ) {
+			// Use a regular variable so we can call a static method from it PHP <7.0
+			$attendee_provider = $this->attendee_provider;
+
+			$args['order_id'] = $attendee_provider::generate_order_id();
+		}
+
+		// Set default attendee status.
+		if ( ! isset( $args['attendee_status'] ) ) {
+			$args['attendee_status'] = 'yes';
+		}
+
+		return $args;
+	}
+
+	/**
+	 * Handle backwards compatible creation actions for RSVPs.
+	 *
+	 * @since TBD
+	 *
+	 * @param WP_Post                       $attendee      The attendee object.
+	 * @param array                         $attendee_data List of additional attendee data.
+	 * @param Tribe__Tickets__Ticket_Object $ticket        The ticket object.
+	 */
+	public function trigger_create_actions( $attendee, $attendee_data, $ticket ) {
+		parent::trigger_create_actions( $attendee, $attendee_data, $ticket );
+
+		$attendee_id       = $attendee->ID;
+		$post_id           = Arr::get( $attendee_data, 'post_id' );
+		$order_id          = $attendee_data['order_id'];
+		$product_id        = $ticket->ID;
+		$order_attendee_id = Arr::get( $attendee_data, 'order_attendee_id' );
```

Comment:
Do we need to set defaults for the array get? Is there any case these can be unset?

Reply:
It's possible it won't be set in cases like MA, but set in cases like the normal RSVP flows